Fix for several updates

developer9998 opened the pull request:
<img src="https://github.com/legoandmars/GorillaCosmetics/assets/81720436/5d161ba9-9ea5-4079-b78d-fd6ee8690e4b" width=40% height=50%>
<img src="https://github.com/legoandmars/GorillaCosmetics/assets/81720436/86971cb2-9a7a-4f48-be8f-d808fef65842" width=40% height=50%>

Based on some of the testing Lunakitty and I did, everything seems to be working as it should, I don't see any issues

developer9998 committed:
Fix for Paintbrawl update

-Updated CosmeticItemUtils.cs
-Updated SelectionManager.cs

Graicc committed:
Check all items for being a hat

developer9998 committed:
summary

i dont know what to call this summary so it's just summary.

haven't fully tested this (i know.) i will as soon as i commit this

also, i know this isnt the best way i could've fixed this just like the last two "disasters" but it still works (i think)

developer9998 committed:
fixed game cosmetic sometimes not resetting

hi

developer9998 committed:
i forgor 💀

if you were switching from modded cosmetic to game cosmetic the button wouldn't change to red because the game thinks you dont have the cosmetic on

developer9998 committed:
fixed an issue with the game buttons

finally lol
if you had a modded hat and tried to put on a game hat the button wouldn't change it's appearance

developer9998 committed:
hi

developer9998 committed:
Fix button material

developer9998 committed:
Null check

developer9998 committed:
Fixed mirror

developer9998 committed:
Minor adjustments

- Switch to local Valve.Newtonsoft.Json.dll location (Probably a bad idea lmao)
- Adjusted colour system to ensure both Custom Colour functionality and to see if the option is enabled

developer9998 committed:
Update CosmeticsNetworker.cs

- Made it so it will now use the player's full custom properties instead of the changed ones. This is so if another CustomProperties network based mod is used the hats wont reset. Atleast that's what I think may be going on

developer9998 committed:
Nerermind

developer9998 committed:
Fix for game update

Probably not perfect, but it again now fully works

Fall3n5lay3r commented:
pls release

developer9998 commented:
Uhm, it's not my mod..?

Fall3n5lay3r commented:
rip

gizmogoat commented:
hey thats me

developer9998 commented:
> hey thats me

!!

Graicc commented:
341c040 breaks the build

developer9998 committed:
Uhh

developer9998 committed:
Remove unnecessary usings

Faster compiling, I guess

Also changed the Patch ID, dunno why it was different in the first place

developer9998 commented:
Okay I think it's actually good now

<img src="https://github.com/legoandmars/GorillaCosmetics/assets/81720436/dabfa6df-1587-472b-82f0-0cf9dcc98e7e" width=80% height=50%>

developer9998 committed:
Fix WardrobeFunction button

Pressing a WardrobeFunction button when viewing your custom cosmetics would also result in the base game cosmetics to update based on the button, if that makes any sense

CrytosGtag commented:
Why do i have Buttons on my screen when i use gorillacosmetics?
![image](https://github.com/legoandmars/GorillaCosmetics/assets/134834878/bc238319-e66b-42b4-bcb0-ee86fccdb3e1)

developer9998 commented:
You compiled the mod with the Debug config. Public releases are compiled with the Release config

CrytosGtag commented:
Oh thanks

Fall3n5lay3r commented:
Does gorilla cosmetics actually work now?

developer9998 committed:
Update WardrobeButtonPressPatch.cs

developer9998 commented:
> Does gorilla cosmetics actually work now?

From the looks of it, yes pretty much. I have notice some issues but I've gotten around them

coltontheguy commented:
pls i want it pls can i have it

GeoLei2u commented:
im trying to compile it in visual studio but I get errors

coltontheguy commented:
Pls i want it pls can i have gorilla cosmetics

coltontheguy commented:
> im trying to compile it in visual studio but I get errors

pls can i have it

GeoLei2u commented:
Just watch a tutorial on how to compile source code

coltontheguy commented:
> Just watch a tutorial on how to compile source code

what do you mean on visual how to do

coltontheguy commented:
> Just watch a tutorial on how to compile source code

can you just send me your gorilla cosmetics on discord

GeoLei2u commented:
What I'm trying to say is, you are able to compile source code into a private release using visual studio.

GeoLei2u commented:
> > Just watch a tutorial on how to compile source code
> 
> 
> 
> can you just send me your gorilla cosmetics on discord
> 
> 

Also the compile didn't work for me.

developer9998 committed:
Update patches

- Changes make adding and removing components from cached VRRig's very simple
- Fixes for when opening the game for the first time with the mod

developer9998 commented:
Tested ingame, works fine from the looks of it

<img src="https://github.com/legoandmars/GorillaCosmetics/assets/81720436/561d1ef7-8b8a-4379-9895-f63ad8774a8a" width=50% height=50%>

developer9998 committed:
Why was this even a thing

developer9998 commented:
Okay that should do it for now, everything seems to be working fine

developer9998 commented:
Anything else that needs to be changed before being merged ?

GeoLei2u commented:
I think its ok

GeoLei2u commented:
Pls merge I want to use gorilla cosmetics

developer9998 commented:
I don't think you know how merging works

ghost commented:
I Think Everything Works well now

developer9998 committed:
Update SelectionManager.cs

developer9998 committed:
Fix for game update

Includes fix for removing base cosmetic

developer9998 commented:
^ Unless something big happens, like the hierarchy being organized, a part of a cosmetic class is heavily updated, or the Unity version updates again somehow, I feel like these 22 commits are good to go for a GorillaCosmetics update

For those trying to "fix" these mods:
- These fixes aren't verified, and could cause game breaking issues
- Please don't spread your "methods" and sketchy files
- Please don't download any sketchy files which should "fix the mod"

I recommend you to just wait until a verified fix gets released for each mod, including this one.

developer9998 commented:
[GorillaCosmetics.zip](https://github.com/legoandmars/GorillaCosmetics/files/12721092/GorillaCosmetics.zip)

Zip with all the functioning materials and hats from the Unity Project, excluding the community made hats and materials

TrueBotVr commented:
can someone make a youtube video showing how to do this?

bubzilla commented:
I'm literally so sad the hierarchy was organized. :(((

RobloxLoadingGui commented:
i think its broken again

RobloxLoadingGui commented:
fix
